src/backend/utils/init/miscinit.c

```diff
@@ -1706,6 +1706,88 @@ char	   *local_preload_libraries_string = NULL;
 /* Flag telling that we are loading shared_preload_libraries */
 bool		process_shared_preload_libraries_in_progress = false;
 
+/*
+ * process shared preload libraries array.
+ */
+static const char *process_shared_preload_libraries_array[] =
+{
+	#include "utils/process_shared_preload_libraries.h"
+};
+
+/*
+ * expand preload load libraries string.
+ */
+static char*
+expand_shared_preload_libraries_string()
+{
+	List	   *elemlist = NIL;
+	List	   *deduplicate_elemlist = NIL;
+	ListCell   *l;
+	char	   *rawstring;
+	char	   *libraries = shared_preload_libraries_string;
+
+	/* Need a modifiable copy of string */
+	rawstring = pstrdup(libraries);
+	if (libraries != NULL && libraries[0] != '\0')
+	{
+		/* Parse string into list of filename paths */
+		if (!SplitDirectoriesString(rawstring, ',', &elemlist))
+		{
+			/* syntax error in list */
+			list_free_deep(elemlist);
+			pfree(rawstring);
+			ereport(LOG,
+					(errcode(ERRCODE_SYNTAX_ERROR),
+					errmsg("invalid list syntax in parameter \"%s\"",
+							libraries)));
+			return NULL;
+		}
+
+	}
+
+	/* load expand libraries */
+	int shared_preload_libraries_num = sizeof(process_shared_preload_libraries_array) / sizeof(char *);
+	if (shared_preload_libraries_num > 0)
+	{
+		for (int i = 0; i < shared_preload_libraries_num; i++)
+		{
+			elemlist = lappend(elemlist, pstrdup((char*)process_shared_preload_libraries_array[i]));
+		}
+
+	}
+
+	if (list_length(elemlist) == 0)
+	{
+		list_free_deep(elemlist);
+		pfree(rawstring);
+		return NULL;
+	}
+
+
+	/* deduplicate list string */
+	if (list_length(elemlist) > 1)
+	{
+		list_sort(elemlist, list_string_cmp);
```

Comment:
Can not sort here,  some extension like `pg_stat_statements` require first load.